File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/executeAction.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Action } from './action';

/**
 * Sharepoint action.execute 
 */
export class ExecuteAction extends Action{
    protected parameters: { [key: string] : unknown };
    private verb: string;

    /**
     * Initializes a new instance of the ExecuteAction class
     */
    public ExecuteAction() {
        // Do nothing
    }

    /**
     * Sets parameters property of type key:value pair
     */
    public set Parameters (parameters: { [key: string] : unknown }) {
        this.parameters = parameters;
    }

    /**
     * Gets parameters property of type key:value pair
     */
    public get Parameters(): { [key: string] : unknown } {
        return this.parameters; 
    }

    /**
     * Sets verb property of type string
     */
    public set Verb (verb: string) {
        this.verb = verb;
    }

    /**
     * Gets verb property of type string
     */
    public get Verb(): string {
        return this.verb; 
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/ExecuteAction.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// Action.Execute.
    /// </summary>
    public class ExecuteAction : Action
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ExecuteAction"/> class.
        /// </summary>
        public ExecuteAction()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the action parameters of type <see cref="Dictionary{TKey, TValue}"/>.
        /// </summary>
        /// <value>This value is the parameters of the action.</value>
        [JsonProperty(PropertyName = "parameters")]
        #pragma warning disable CA2227
        public new Dictionary<string, object> Parameters { get; set; }

        /// <summary>
        /// Gets or Sets the verb associated with this action of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the verb associated with the action.</value>
        [JsonProperty(PropertyName = "verb")]
        public string Verb { get; set; }
    }
}
